libraries/xcode.rb

```diff
@@ -1,5 +1,5 @@
-module Xcode
-  module Helper
+module MacOS
+  module XcodeHelpers
```

Comment:
IMO the word `Helpers` is just needless clutter in these modules. It adds no useful information at the call site.

Reply:
I agree - this module's scope is pretty big. Any name ideas?

Reply:
Like @ehanlon said, the best thing we could do is to extract appropriate methods to the `Chef::Provider.action_class`. The size/scope of this module will also be reduced in the `feature/xcode-idempotence` branch.

Reply:
I like @brownmike's suggestion of using a module's static method - `Xcode.installed?` rather than `xcode_installed`.

Reply:
Since this is a pretty high-level library cookbook, I'd prefer we stick with idiomatic Chef and make contributions to Chef core if we think their current design modal can be improved.

Reply:
https://docs.chef.io/libraries.html uses the following custom namespace example for an `ISP` class:

```
ISP.vhosts.each do |vhost|
  directory vhost[:documentroot] do
    owner 'vhost[:uid]'
    group 'vhost[:gid]'
    mode '0755'
    action :create
  end
```
This change would put us more in line with the Chef docs, in my opinion.